packages/mdc-select/foundation.ts

```diff
@@ -87,15 +92,41 @@ export class MDCSelectFoundation extends MDCFoundation<MDCSelectAdapter> {
     this.helperText_ = foundationMap.helperText;
   }
 
-  setSelectedIndex(index: number) {
-    this.adapter_.setSelectedIndex(index);
-    this.adapter_.closeMenu();
+  /** Returns the index of the currently selected menu item, or -1 if none. */
+  getSelectedIndex(): number {
+    const selectedMenuItem = this.adapter_.getSelectedMenuItem();
+    return selectedMenuItem ? this.adapter_.getMenuItems().indexOf(selectedMenuItem) : -1;
+  }
+
+  setSelectedIndex(index: number, closeMenu = false) {
+    const menuItemToSelect = this.adapter_.getMenuItems()[index];
+    this.adapter_.setSelectedText(
+        menuItemToSelect ? this.adapter_.getMenuItemText(menuItemToSelect).trim() : '');
+
+    const previouslySelectedItem = this.adapter_.getSelectedMenuItem();
+    if (previouslySelectedItem) {
+      this.adapter_.toggleMenuItemSelectedClass(previouslySelectedItem, false);
+      // Remove the ARIA_SELECTED attribute.
+      this.adapter_.setAttributeForElement(previouslySelectedItem, strings.ARIA_SELECTED_ATTR);
+    }
+    if (menuItemToSelect) {
+      this.adapter_.toggleMenuItemSelectedClass(menuItemToSelect, true);
```

Comment:
Same here. This adapter method can refer Index.

We can use even more generic adapter for future usecases where we need to add another class name:

`addClassAtIndex(index: number, className: string)`

`removeClassAtIndex(index: number, className: string)`